### PR TITLE
1.20.2 + NeoForge port

### PR DESCRIPTION
**Pull request by ChrysanthCow**
There's quite a bit here that I'd have to get through if I were to talk about this in depth. The main changes you'll want to look at are the recipe related code (due to rewrites in 1.20.2).

### CODE REVIEW

**Comment by ChrysanthCow** [comment]
Converting this to a draft until Neo gets its registry code overhaul and this has been updated.

**Comment by ChrysanthCow** [comment]
Because the Capability and Networking reworks are planned for 1.20.3 instead, I'm going to mark this PR as ready.

I'll be updating this to 1.20.3 if a NeoForge version releases there prior to this being merged.

File: src/main/resources/META-INF/mods.toml [hunk 1]
```diff
@@ -18,16 +18,16 @@ itemIcon = "farmersdelight:stove"
...
-versionRange = "[47.1.0,)"
+versionRange = "[1,)"
 ordering = "NONE"
 side = "BOTH"
 
 [[dependencies.farmersdelight]]
 modId = "minecraft"
 mandatory = true
-versionRange = "[1.20,)"
+versionRange = "[1.20.2,)"
```
Comment (Witixin1512) [inline src/main/resources/META-INF/mods.toml]: The version range should exclude 1.20.3 in my opinion, as the chances of the mod working out of the box are very low.

Comment (ChrysanthCow) [reply src/main/resources/META-INF/mods.toml]: Agreed. I know for a fact that it won't work because blocks require codecs in init now. (Preparation for data driven blocks I think?)

File: src/generated/resources/data/farmersdelight/recipes/cutting/acacia_door.json [hunk 2]
```diff
@@ -10,8 +10,24 @@
       "item": "minecraft:acacia_planks"
     }
   ],
-  "tool": {
-    "type": "farmersdelight:tool_action",
-    "action": "axe_dig"
-  }
+  "tool": [
```
Comment (vectorwing) [inline src/generated/resources/data/farmersdelight/recipes/cutting/acacia_door.json]: Is there a reason why `tool_action` was removed on those recipes?
Hardcoding the tools like this will prevent any modded tool from functioning; that's what the `tool_action` ingredient is for.

File: src/main/java/vectorwing/farmersdelight/common/block/BuddingTomatoBlock.java [hunk 3]
```diff
@@ -73,4 +74,9 @@ public void performBonemeal(ServerLevel level, RandomSource random, BlockPos pos
 			level.setBlockAndUpdate(pos, ModBlocks.TOMATO_CROP.get().defaultBlockState().setValue(TomatoVineBlock.VINE_AGE, remainingGrowth));
 		}
 	}
+
+	@Override
+	public boolean isEnabled(FeatureFlagSet p_249172_) {
```
Comment (vectorwing) [inline src/main/java/vectorwing/farmersdelight/common/block/BuddingTomatoBlock.java]: Mostly a question: is this method related to the experimental feature flags? If so, why was it added to BuddingTomatoBlock only?

**Comment by vectorwing** [comment]
Hello! Thank you very much for this whole port, and sorry for taking so long to reply to you. You caught me during a long modding break, so I didn't get to this in time. 😅

I took a general look over the changes, and asked other modders in regards to the NeoForge porting bits; so far, it seems good! I'll try cloning it locally to run the port later. There are a few things I'd like to discuss before approval, though; I started a review for them.

Now, since this is migrating FD to a new Minecraft version, I need to create a new branch for it, which I'll redirect this PR to manually once I do it. We'll keep the 1.20.1 branch as-is.

Also... after asking around, I think I'll be jumping directly to 1.20.4 rather than 1.20.2, given general mod support and NF refactors. My plan is to merge this PR, then do the work to bring it to 1.20.4 later.

Lemme know if you're available on Discord or similar, if you'd like to discuss the PR further. 👍

File: src/main/java/vectorwing/farmersdelight/integration/jei/FDRecipes.java [hunk 4]
```diff
@@ -1,5 +1,7 @@
 package vectorwing.farmersdelight.integration.jei;
 
+// TODO: Uncomment this once JEI updates.
```
Comment (vectorwing) [inline src/main/java/vectorwing/farmersdelight/integration/jei/FDRecipes.java:3]: Due to how long I took to review this, both JEI and CT have already ported to both 1.20.2 and 1.20.4... Sorry about that. x)

**Review by vectorwing — CHANGES_REQUESTED** [review]

File: src/generated/resources/data/farmersdelight/recipes/cutting/acacia_door.json (hunk 2 above)
Comment (ChrysanthCow) [reply src/generated/resources/data/farmersdelight/recipes/cutting/acacia_door.json]: I'll look into it.

File: src/main/java/vectorwing/farmersdelight/common/block/BuddingTomatoBlock.java (hunk 3 above)
Comment (ChrysanthCow) [reply src/main/java/vectorwing/farmersdelight/common/block/BuddingTomatoBlock.java]: That is incredibly weird. Will fix at a later date.

**Comment by ChrysanthCow** [comment]
Would you like me to update this to 1.20.4 after I fix the above issues? Thanks for telling me about these.

Edit: Ah, I read what you said, and you're willing to do the work. Thanks for letting me know.

File: src/main/java/vectorwing/farmersdelight/integration/jei/FDRecipes.java (hunk 4 above)
Comment (ChrysanthCow) [reply src/main/java/vectorwing/farmersdelight/integration/jei/FDRecipes.java:3]: I've updated CraftTweaker, however, JEI is not available on NeoForge for 1.20.2, but is on 1.20.4. You'll have to handle JEI.

File: src/main/java/vectorwing/farmersdelight/common/block/BuddingTomatoBlock.java (hunk 3 above)
Comment (ChrysanthCow) [reply src/main/java/vectorwing/farmersdelight/common/block/BuddingTomatoBlock.java]: Fixed.

File: src/generated/resources/data/farmersdelight/recipes/cutting/acacia_door.json (hunk 2 above)
Comment (ChrysanthCow) [reply src/generated/resources/data/farmersdelight/recipes/cutting/acacia_door.json]: Fixed, I've explained in Discord DMs what the fix was.

**Review by vectorwing — APPROVED** [review]

**Comment by vectorwing** [comment]
Looking good! I managed to resolve all the conflicts; turns out they were mostly imports and removed `@Dist` tags.
Thank you again for all the work on this port! I appreciate it. ^^
